pkg/nsx/services/realizestate/realize_state.go

```diff
@@ -49,7 +49,7 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 			return err
 		}
 		for _, result := range results.Results {
-			if *result.EntityType != entityType {
+			if entityType != "" && *result.EntityType != entityType {
```

Comment:
better to check if result.EntityType is nil

Reply:
Fixed.